pkg/kubelet/metrics/metrics.go

```diff
@@ -126,8 +127,30 @@ const (
 	EphemeralContainer = "ephemeral_container"
 )
 
+type imageSizeBucket struct {
+	lowerBoundInBytes uint64
+	label             string
+}
+
 var (
 	podStartupDurationBuckets = []float64{0.5, 1, 2, 3, 4, 5, 6, 8, 10, 20, 30, 45, 60, 120, 180, 240, 300, 360, 480, 600, 900, 1200, 1800, 2700, 3600}
+	// imageSizeBuckets has the labels to be associated with image_pull_duration_seconds metric. For example, if the size of
```

Comment:
why do you use labels instead the histogram directly?

Reply:
because we are trying to associate each latency with its image size.

With this CL, we will have metric datapoints like these:

{10 seconds, "LE10MB"}, {100 seconds, "GT100MB"}, {240 seconds, "GT5GB"}, {270 seconds, "GT5GB"}.

And then we will be able to filter the datapoints by image size, to know how we are performing for large images or small images.

Histogram without labels can only give us the distribution of the latency, across all image sizes, and it makes the metric pretty unactionable.